### ISSUE TITLE
Can't use carrierwave without fog after #1620

### ISSUE THREAD

**Issue by varyform** [carrierwaveuploader/carrierwave#1622]
I'm using carrierwave with https://github.com/sorentwo/carrierwave-aws instead of fog. 
After #1620 CW fails to load when fog is not installed/bundled.

**Comment by ktaras** [comment]
:+1: I have exactly the same problem!

**Comment by plribeiro3000** [comment]
Can you guys please provide the `carrierwave` config and the error you guys are getting?

**Comment by ktaras** [comment]
here is the error:
```
$ rails s
/Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require': cannot load such file -- fog (LoadError)
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `block in require'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:240:in `load_dependency'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader/configuration.rb:112:in `eager_load_fog'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader/configuration.rb:127:in `fog_credentials='
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader/configuration.rb:166:in `block in reset_config'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader/configuration.rb:149:in `configure'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader/configuration.rb:156:in `reset_config'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader/configuration.rb:45:in `block in <module:Configuration>'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/concern.rb:120:in `class_eval'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/concern.rb:120:in `append_features'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader.rb:48:in `include'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader.rb:48:in `<class:Base>'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader.rb:45:in `<module:Uploader>'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader.rb:28:in `<module:CarrierWave>'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave/uploader.rb:23:in `<top (required)>'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `block in require'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:240:in `load_dependency'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/bundler/gems/carrierwave-186e89055821/lib/carrierwave.rb:92:in `<top (required)>'
from /Users/taras/.rvm/gems/ruby-2.2.0@global/gems/bundler-1.7.12/lib/bundler/runtime.rb:76:in `require'
from /Users/taras/.rvm/gems/ruby-2.2.0@global/gems/bundler-1.7.12/lib/bundler/runtime.rb:76:in `block (2 levels) in require'
from /Users/taras/.rvm/gems/ruby-2.2.0@global/gems/bundler-1.7.12/lib/bundler/runtime.rb:72:in `each'
from /Users/taras/.rvm/gems/ruby-2.2.0@global/gems/bundler-1.7.12/lib/bundler/runtime.rb:72:in `block in require'
from /Users/taras/.rvm/gems/ruby-2.2.0@global/gems/bundler-1.7.12/lib/bundler/runtime.rb:61:in `each'
from /Users/taras/.rvm/gems/ruby-2.2.0@global/gems/bundler-1.7.12/lib/bundler/runtime.rb:61:in `require'
from /Users/taras/.rvm/gems/ruby-2.2.0@global/gems/bundler-1.7.12/lib/bundler.rb:134:in `require'
from /Users/taras/Documents/rg/zenit/zenit/config/application.rb:7:in `<top (required)>'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:78:in `require'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:78:in `block in server'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:75:in `tap'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:75:in `server'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
from /Users/taras/.rvm/gems/ruby-2.2.0@app/gems/railties-4.2.1/lib/rails/commands.rb:17:in `<top (required)>'
from bin/rails:4:in `require'
from bin/rails:4:in `<main>'
```

**Comment by plribeiro3000** [comment]
@ktaras What about your `Gemfile`? Does it have `fog` or `fog-aws`?

**Comment by ktaras** [comment]
No, just
```
gem 'carrierwave', github: 'carrierwaveuploader/carrierwave'
gem 'carrierwave-aws'
```
before update it had been working perfectly

**Comment by plribeiro3000** [comment]
Can you provide your `carrierwave` configurationa s well?

**Comment by ktaras** [comment]
@plribeiro3000 And here is config:
```
if Rails.env.test?
  CarrierWave.configure do |config|
    config.storage = :file
    config.enable_processing = false
  end
else
  CarrierWave.configure do |config|
    config.storage = :aws
    config.cache_storage = :aws
    config.aws_credentials = {
      access_key_id:     Rails.application.secrets.amazon_access_key_id,
      secret_access_key: Rails.application.secrets.amazon_access_key_secret,
      region:            Rails.application.secrets.amazon_creatives_region
    }
    config.aws_acl       = :public_read
    config.aws_bucket    = Rails.application.secrets.amazon_creatives_dir
    config.store_dir     = nil # store at the root level
    config.cache_dir     = 'cache'
    config.asset_host    = Proc.new do |source|
      Rails.application.secrets.creatives_asset_host unless source.cached?
    end
  end
end
```

**Comment by plribeiro3000** [comment]
So the issue is in this [line](https://github.com/carrierwaveuploader/carrierwave/blob/014c091fae86d6439cafd432473e3cc026be11f1/lib/carrierwave/uploader/configuration.rb#L166) that triggers this [line](https://github.com/carrierwaveuploader/carrierwave/blob/014c091fae86d6439cafd432473e3cc026be11f1/lib/carrierwave/uploader/configuration.rb#L110).

Gotta send a fix for this soon.

**Comment by ktaras** [comment]
@plribeiro3000 thank you!

**Comment by maninalift** [comment]
I had this issue when using `fog-aws` with carrierwave and worked around it by using the `fog` gem rather than just `fog-aws`.

I'm not sure whether it is related but I also had to remove the line

```ruby
config.fog_provider = 'fog-aws' 
```

in my config when compared to the configuration example given in the "Using Amazon S3" section of the readme.

**Comment by eavgerinos** [comment]
This issue is fixed. Please check the changes in README to re-configure your applications.

**Comment by estebanbouza** [comment]
It seems that I'm still having this issue. Here's the error

```
$ rails s
/Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require': cannot load such file -- fog (LoadError)
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `block in require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/carrierwave-0.10.0/lib/carrierwave/storage/fog.rb:3:in `<top (required)>'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `block in require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/carrierwave-0.10.0/lib/carrierwave/storage.rb:9:in `<top (required)>'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `block in require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/carrierwave-0.10.0/lib/carrierwave.rb:78:in `<top (required)>'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.9.2/lib/bundler/runtime.rb:76:in `require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.9.2/lib/bundler/runtime.rb:76:in `block (2 levels) in require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.9.2/lib/bundler/runtime.rb:72:in `each'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.9.2/lib/bundler/runtime.rb:72:in `block in require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.9.2/lib/bundler/runtime.rb:61:in `each'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.9.2/lib/bundler/runtime.rb:61:in `require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bundler-1.9.2/lib/bundler.rb:134:in `require'
	from /Users/user/Dropbox/myprojects/loyalty/loyalty/config/application.rb:7:in `<top (required)>'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:78:in `require'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:78:in `block in server'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:75:in `tap'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:75:in `server'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/user/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:8:in `require'
	from bin/rails:8:in `<main>'
```

My Gemfile:
```
gem 'fog-aws'
gem 'carrierwave'
```


My related Gemfile.lock 

```
    fog-aws (0.4.0)
      fog-core (~> 1.27)
      fog-json (~> 1.0)
      fog-xml (~> 0.1)
      ipaddress (~> 0.8)
    fog-core (1.30.0)
      builder
      excon (~> 0.45)
      formatador (~> 0.2)
      mime-types
      net-scp (~> 1.1)
      net-ssh (>= 2.1.3)
    fog-json (1.0.1)
      fog-core (~> 1.0)
      multi_json (~> 1.0)
    fog-xml (0.1.2)
      fog-core
      nokogiri (~> 1.5, >= 1.5.11)
```

I'm not including my initializer configuration as the error happens regardless of the configuration I use.

Any ideas on what can be going wrong?

**Comment by eavgerinos** [comment]
@estebanbouza  You are using carrierwave version 0.10.0. The documentation in the README of master branch refers to the master branch. Please, for documentation see [here](https://github.com/carrierwaveuploader/carrierwave/tree/v0.10.0).

**Comment by estebanbouza** [comment]
Working now with the following Gemfile
```
gem 'fog-aws'
gem 'carrierwave', github: 'carrierwaveuploader/carrierwave'
```

Thanks! 👍

**Comment by eavgerinos** [comment]
Anytime :)

**Comment by andiruleu** [comment]
Just tried to do the same, did what @estebanbouza did. Originally didn't have carrierwave, I guess we need this gem to install the mailboxes gem?

**Comment by thomasfedb** [comment]
Please create a new issue report. This issue is many months old.
